File: Problemas sobre RSQ y LIS/spiderman.py
from sys import stdin

def jump_buildings(n, heights):
    min_energy = [float("inf")] * n
    min_energy[0] = 0
    for i in range(n):
        # Itero sobre las potencias de 2
        jump = 1
        while (i + jump) < n:
            # Edificio al que salta
            j = i + jump
            energy = abs(heights[j] - heights[i])
            # Mínimo entre la energía que tenía y la nueva energía + lo que cuesta llegar al edificio donde estoy
            min_energy[j] = min(min_energy[j], energy + min_energy[i])
            jump *= 2
    return min_energy[n - 1]


input_data = stdin.read().strip().splitlines()

for i in range(0, len(input_data), 2):
    n = int(input_data[i])
    heights = list(map(int, input_data[i + 1].split()))
    print(jump_buildings(n, heights))

File: Problemas sobre RSQ y LIS/spiderman.js
const jumpBuildings = (n, heights) => {
    const minEnergy = new Array(n).fill(Infinity);
    minEnergy[0] = 0;
    for (let i = 0; i < n; i++) {
        // Itero sobre las potencias de 2
        for (let jump = 1; (i + jump < n); jump *= 2) {
            // Edificio al que salta
            const j = i + jump;
            const energy = Math.abs(heights[j] - heights[i]);
            // Mínimo entre la energía que tenía y la nueva energía + lo que cuesta llegar al edificio donde estoy
            minEnergy[j] = Math.min(minEnergy[j], energy + minEnergy[i]);
        }
    }
    return minEnergy[n - 1];
}

const input = require('fs').readFileSync(0, 'utf-8').trim().split('\n');

for (let i = 0; i < input.length; i += 2) {
    const n = parseInt(input[i]);
    const heights = input[i + 1].split(' ').map(Number);
    console.log(jumpBuildings(n, heights));
}